assets/src/design-system/components/tabGroup/index.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { forwardRef, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Tab } from '../tab';
+
+const StyledTabGroup = styled.div`
+  display: flex;
+  align-items: flex-start;
+  flex-wrap: wrap;
+
+  & > button {
+    margin-left: 8px;
+    &:first-of-type {
+      margin-left: 0;
+    }
+  }
+`;
+
+export const TabGroup = forwardRef(function TabGroup(
+  {
+    getTabId = (id) => id,
+    getAriaControlsId,
+    tabs,
+    label,
+    handleTabClicked,
+    activeTabId,
+  },
+  ref
+) {
+  const tabRefs = useRef({});
+
+  return (
+    <StyledTabGroup ref={ref} aria-label={label} role="tablist">
+      {tabs.map(({ id, title, ...rest }) => {
+        const isActive = activeTabId === id;
+        return (
+          <Tab
+            key={`${title.replace(/\s/g, '')}_${id}`}
```

Comment:
One little perf comment: Let's hoist this regex literal since generating it each map can be expensive.